ctapipe/tools/muon_reconstruction.py

```diff
@@ -5,121 +5,223 @@
 The resulting output can be read e.g. using for example
 `pandas.read_hdf(filename, 'muons/LSTCam')`
 """
-
-import warnings
-from collections import defaultdict
-
 from tqdm import tqdm
+import numpy as np
+from astropy.coordinates import SkyCoord
+from ctapipe.containers import TelEventIndexContainer
 
 from ctapipe.calib import CameraCalibrator
 from ctapipe.core import Provenance
 from ctapipe.core import Tool, ToolConfigurationError
-from ctapipe.core import traits as t
-from ctapipe.image.muon.muon_diagnostic_plots import plot_muon_event
-from ctapipe.image.muon.muon_reco_functions import analyze_muon_event
+from ctapipe.core.traits import Unicode, IntTelescopeParameter
 from ctapipe.io import EventSource
 from ctapipe.io import HDF5TableWriter
+from ctapipe.image.cleaning import TailcutsImageCleaner
+from ctapipe.coordinates import TelescopeFrame, CameraFrame
+from ctapipe.image.muon import MuonRingFitter, MuonIntensityFitter, ring_containment
 
-warnings.filterwarnings("ignore")  # Supresses iminuit warnings
-
-
-def _exclude_some_columns(subarray, writer):
-    """ a hack to exclude some columns of all output tables here we exclude
-    the prediction and mask quantities, since they are arrays and thus not
-    readable by pandas.  Also, prediction currently is a variable-length
-    quantity (need to change it to be fixed-length), so it cannot be written
-    to a fixed-length table.
-    """
-    all_camids = {str(x.camera) for x in subarray.tel.values()}
-    for cam in all_camids:
-        writer.exclude(cam, 'prediction')
-        writer.exclude(cam, 'mask')
 
-
-class MuonDisplayerTool(Tool):
+class MuonAnalysis(Tool):
     name = 'ctapipe-reconstruct-muons'
-    description = t.Unicode(__doc__)
+    description = Unicode(__doc__)
 
-    outfile = t.Unicode(
-        "muons.hdf5",
+    output = Unicode(
+        default_value=None,
+        allow_none=True,
         help='HDF5 output file name'
     ).tag(config=True)
 
-    display = t.Bool(
-        help='display the camera events', default=False
+    min_pixels = IntTelescopeParameter(
+        help=(
+            'Minimum number of pixels after cleaning and ring finding'
+            'required to process an event'
+        ),
+        default_value=100,
     ).tag(config=True)
 
-    classes = t.List([
-        CameraCalibrator, EventSource
-    ])
+    classes = [
+        CameraCalibrator, EventSource, MuonRingFitter, MuonIntensityFitter,
+    ]
 
-    aliases = t.Dict({
+    aliases = {
+        'i': 'EventSource.input_url',
         'input': 'EventSource.input_url',
-        'outfile': 'MuonDisplayerTool.outfile',
-        'display': 'MuonDisplayerTool.display',
+        'o': 'MuonAnalysis.output',
+        'output': 'MuonAnalysis.output',
         'max_events': 'EventSource.max_events',
         'allowed_tels': 'EventSource.allowed_tels',
```

Comment:
In other tools, I've used "max-events" and "allowed-tels" for these (following usual command-line arg conventions of not using underscores). But that's not critical. In the future we should unify all tool arguments, though.